python_modules/dagster/dagster/_core/pipes/context.py

```diff
@@ -275,6 +276,40 @@ class PipesSession:
     message_reader_params: PipesParams
     context: OpExecutionContext
 
+    @property
+    def dagster_tags(self) -> Dict[str, str]:
+        """Important Dagster key-value pairs typically attached to external resources launched from Pipes."""
```

Comment:
A more descriptive name and docblock would be helpful.

Maybe:

`default_remote_invocation_tags`

```python
"""
Remote execution environments commonly have their own concepts of tags or labels. It is useful to attach dagster-specific metadata. For example the out-of-the-box PipesK8Client uses this to attach tags to any pod launched by the client, so those tags appear in Kubernetes-native tooling.

By default the tags returns are:

* ...
"""
```

Also we should document the exhaustive list of string keys somewhere.

Reply:
Maybe `@cached_property` here since there is a bunch of logic?